src/blosc2/schunk.py

```diff
@@ -725,6 +875,26 @@ def __setitem__(self, key, value):
         This method can also be used to append new data if :paramref:`key`.stop
         is greater than the SChunk nitems.
 
+        Examples
+        --------
+        >>> import blosc2
+        >>> import numpy as np
+        >>> nchunks = 4
+        >>> chunk_size = 200 * 1000 * 4
+        >>> data = np.arange(nchunks * chunk_size // 4, dtype=np.int32)
+        >>> schunk = blosc2.SChunk(chunksize=chunk_size, data=data, cparams={"typesize": 4})
+        >>> # Create a new array of values to update the slice (values from 1000 to 1999 multiplied by 2)
+        >>> start_ = 1000
+        >>> stop = 2000
+        >>> new_values = np.arange(start_, stop, dtype=np.int32) * 2
+        >>> # Use __setitem__ by the slicing syntax to update the chunk
+        >>> schunk[start_:stop] = new_values
+        >>> # Retrieve the updated slice using the slicing syntax
+        >>> retrieved_slice = np.frombuffer(schunk[start_:stop], dtype=np.int32)
+        >>> # Compare the retrieved slice with the new values to ensure they match
+        >>> assert np.array_equal(retrieved_slice, new_values), "The updated slice does not match the new values."
+        >>> print("The slice comparison is successful!")
+        The slice comparison is successful!
```

Comment:
Ditto.  By removing the assert here, we can eliminate the last part of the example.  If anything, you may want to print some values of the modified part; IMO examples are more clear when people see real data printed.